webapp/javascript/components/FlameGraph/FlameGraphComponent/Highlight.tsx

```diff
@@ -18,17 +18,26 @@ export interface HighlightProps {
   canvasRef: React.RefObject<HTMLCanvasElement>;
 }
 export default function Highlight(props: HighlightProps) {
-  const { canvasRef, barHeight, xyToHighlightData } = props;
+  const { canvasRef, barHeight, xyToHighlightData, zoom } = props;
   const [style, setStyle] = React.useState<React.CSSProperties>({
     height: '0px',
     visibility: 'hidden',
   });
 
+  React.useEffect(() => {
+    if (!zoom) return;
```

Comment:
I believe this branch is never reached! Since `zoom` is an `Option`, which is never falsy.

So what this code is actually doing is EVERY time we change the zoom, we stop highlighting.
Which makes sense IMO, after all, the issue (stale highlighting) happens when BOTH ZOOMING and UNZOOMING.

Therefore I don't think this `if` is necessary.

Reply:
comments added, pretty much what you said